yt/frontends/stream/fields.py

```diff
@@ -87,18 +89,43 @@ class StreamFieldInfo(FieldInfoContainer):
 
     def setup_fluid_fields(self):
         from yt.fields.magnetic_field import setup_magnetic_field_aliases
+        from yt.fields.species_fields import setup_species_fields
+        from yt.utilities.periodic_table import periodic_table
 
+        # First grab all the element symbols from the periodic table
+        symbols = list(periodic_table.elements_by_symbol.keys())
+        # Now add the electron, and some common molecules
+        symbols += ["El", "H2", "CO"]
+        species_names = []
         for field in self.ds.stream_handler.field_units:
             if field[0] in self.ds.particle_types:
                 continue
             units = self.ds.stream_handler.field_units[field]
             if units != "":
                 self.add_output_field(field, sampling_type="cell", units=units)
```

Comment:
Unrelated, but does anyone know why this isn't `sampling_type="local"`? Isn't that what the hexahedral mesh and SPH particles need?

Reply:
^^ @matthewturk

Reply:
So I'm not 100% sure, but as I am reading this I think I believe that at some point cell became local, but that does not actually seem to be the case.  The docs for `DerivedField` are out of date, as they only list `cell`, and `discrete`/`particle` (as synonyms).  Is the distinction related to conservation issues, especially mass vs volume?

Reply:
My recollection is that `"local"` allows us to define gas fields with a spatial extent whether they are cells with a spacing, SPH particles with a smoothing length, etc.

Reply:
Yeah, I generally use `'local'` in trident to deal with both grid- and particle-based codes as per Nathan's suggestions when 4.0 was being developed.

Reply:
I'll most likely address this in a separate PR after I test a few things.